### PR TITLE
Auto load audit

### PR DESCRIPTION
**Pull request by shan-c**

### CODE REVIEW

File: contracts/ERC20_AUTOLOAD_SIMPLE.sol [hunk 1]
```diff
@@ -74,21 +73,31 @@ contract CypherAutoLoad is Pausable, AccessControl, ReentrancyGuard {
...
         _unpause();
     }
 
+    function safeTransfer(
+        IERC20 token,
+        address userAddress,
+        address beneficiaryAddress,
+        uint amount
+    ) external {
+        token.safeTransferFrom(userAddress, beneficiaryAddress, amount);
```
Comment (shan-c) [inline contracts/ERC20_AUTOLOAD_SIMPLE.sol:98]: safeTransferFrom doesn't return any boolean value, hence cannot be used in require. Moved it to an external function to wrap it in try catch

File: contracts/ERC20_AUTOLOAD_SIMPLE.sol [hunk 2]
```diff
@@ -115,18 +124,17 @@ contract CypherAutoLoad is Pausable, AccessControl, ReentrancyGuard {
...
-            "Insufficient funds or allowance"
-        );
-        require(
-            token.transferFrom(userAddress, beneficiaryAddress, amount),
-            "Token transfer failed"
-        );
-        emit Withdraw(tokenAddress, userAddress, beneficiaryAddress, amount);
 
-        return true;
+        try this.safeTransfer(token, userAddress, beneficiaryAddress, amount) {
```
Comment (shan-c) [inline contracts/ERC20_AUTOLOAD_SIMPLE.sol:128]: new external function

File: contracts/ERC20_AUTOLOAD_SIMPLE.sol [hunk 3]
```diff
@@ -74,21 +73,31 @@ contract CypherAutoLoad is Pausable, AccessControl, ReentrancyGuard {
     function setMaxWithdrawalLimit(
         uint newLimit
     ) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        _maxWithdrawalLimit = newLimit;
+        _maxWithdrawalLimitPerTransaction = newLimit;
+        emit MaxWithdrawalLimit(_msgSender(), newLimit);
```
Comment (shan-c) [inline contracts/ERC20_AUTOLOAD_SIMPLE.sol:77]: emits an event on modifying withdrawal limit

File: contracts/ERC20_AUTOLOAD_SIMPLE.sol [hunk 4]
```diff
@@ -21,6 +23,8 @@ contract CypherAutoLoad is Pausable, AccessControl, ReentrancyGuard {
         uint amount
     );
 
+    event MaxWithdrawalLimit(address indexed caller, uint newLimit);
```
Comment (shan-c) [inline contracts/ERC20_AUTOLOAD_SIMPLE.sol:26]: new event for modifying withdrawal limit

File: contracts/ERC20_AUTOLOAD_SIMPLE.sol [hunk 5]
```diff
@@ -115,18 +124,17 @@ contract CypherAutoLoad is Pausable, AccessControl, ReentrancyGuard {
         require(userAddress != address(0), "Invalid user address");
 
         IERC20 token = IERC20(tokenAddress);
-        uint allowed = token.allowance(userAddress, address(this));
-        uint balance = token.balanceOf(userAddress);
-        require(
-            balance >= amount && allowed >= amount,
-            "Insufficient funds or allowance"
-        );
```
Comment (shan-c) [inline contracts/ERC20_AUTOLOAD_SIMPLE.sol:123]: removed this check as it already happens in ERC20 transfer call

File: contracts/ERC20_AUTOLOAD_SIMPLE.sol [hunk 6]
```diff
@@ -46,19 +46,18 @@ contract CypherAutoLoad is Pausable, AccessControl, ReentrancyGuard {
...
         require(
-            amount <= _maxWithdrawalLimit * decimalFactor,
+            amount <= _maxWithdrawalLimitPerTransaction * decimalFactor,
             "Withdrawal amount exceeds limit"
         );
         _;
     }
 
-    modifier anyPrivilagedUser() {
+    modifier anyPrivilegedUser() {
```
Comment (shan-c) [inline contracts/ERC20_AUTOLOAD_SIMPLE.sol:60]: typo fix

File: contracts/ERC20_AUTOLOAD_SIMPLE.sol (hunk 2 above)
Comment (cysedhu) [reply contracts/ERC20_AUTOLOAD_SIMPLE.sol:128]: no need of try catch

**Pull request merged**